dts/arm/rpi_pico/rp2040.dtsi

```diff
@@ -100,6 +100,26 @@
 			status = "disabled";
 		};
 
+		spi0: spi@4003c000 {
+			compatible = "arm,pl022";
```

Comment:
SPI should also be a reset device. @andrei-edward-popa , I suppose the correct way to do this is to create a separate DTS binding for rpi_pico, and have it inherit pl022 and reset-device?

Reply:
@soburi @yonsch Yes, you can create a DTS bindings under dts/bindings/spi/raspberrypi,pico-spi.yaml and inherit arm,pl022.yaml and reset,device.yaml (see raspberrypi,pico-uart.yaml). After that, add resets field inside spi nodes (take a look inside rpi_pico_common.dtsi for spi reset values).

Reply:
Right, you don't need another implementation. Just make sure that enabling SPI in DTS still selects the PL022 implementation

Reply:
I think you can remove arm,pl022 from the compatible strings because raspberrypi,pico-spi yaml inherits arm,pl022.yaml.

Reply:
> I think you can remove arm,pl022 from the compatible string because raspberrypi,pico-spi yaml inherits arm,pl022.yaml.

If do so, It couldn't resolve pl022 in DT_INST_FOREACH_STATUS_OKAY, can't compile it.
It seem to need it.

Compiling with CONFIG_SPI_PL022, I get such as,

```
/home/crs/zephyrproject/zephyr/drivers/spi/spi_pl022.c:473:12: warning: 'spi_pl022_init' defined but not used [-Wunused-function]
  473 | static int spi_pl022_init(const struct device *dev)
      |            ^~~~~~~~~~~~~~
/home/crs/zephyrproject/zephyr/drivers/spi/spi_pl022.c:468:30: warning: 'spi_pl022_api' defined but not used [-Wunused-variable]
  468 | static struct spi_driver_api spi_pl022_api = {
      |                              ^~~~~~~~~~~~~
```

Reply:
Okay, so probably that's the right way.